src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -279,6 +285,36 @@ public function addClassResource(\ReflectionClass $class)
     }
 
     /**
+     * Retrieves the requested reflection class and registers it for resource tracking.
+     *
+     * @param string $class
+     *
+     * @return \ReflectionClass|false
```

Comment:
Well, if this was a private API, I would be OK with this argument about simplifying the internal implementation.
But this is a public API.